radix-engine-constants/src/lib.rs

```diff
@@ -38,7 +38,7 @@ pub const MAX_TRANSACTION_SIZE: usize = 1 * 1024 * 1024;
 pub const DEFAULT_SYSTEM_LOAN: u32 = 1_000_000;
 
 /// The default max call depth, used by transaction executor.
-pub const DEFAULT_MAX_CALL_DEPTH: usize = 12;
+pub const DEFAULT_MAX_CALL_DEPTH: usize = 10;
```

Comment:
@talekhinezh @iamyulong  A little unsure as to why this was needed, or whether there is a larger problem here. Without reducing this, it seemed like the execution environment stack was overflowing before we reach the `MAX_CALL_DEPTH` and therefore it was erroring out. I should note that this issue goes away when running the tests with `--release`.

The tests that were failing are these two:
* [`recursion_bomb_to_failure`](https://github.com/radixdlt/radixdlt-scrypto/blob/7d1030689f5a0e46e5fc8cafc9434f4cc7b1c958/radix-engine/tests/local_component.rs#L164-L203)
* [`recursion_bomb_2_to_failure`](https://github.com/radixdlt/radixdlt-scrypto/blob/7d1030689f5a0e46e5fc8cafc9434f4cc7b1c958/radix-engine/tests/local_component.rs#L240-L279)

Reply:
Thanks! Will look into this at a later time.

Reply:
Yup, I've run into this problem as well when running in debug mode